common/validation/src/main/kotlin/com/github/triplet/gradle/common/validation/Validation.kt

```diff
@@ -10,15 +10,15 @@ import org.gradle.api.logging.Logger
 import org.gradle.kotlin.dsl.apply
 import org.gradle.kotlin.dsl.findByType
 import org.gradle.kotlin.dsl.withType
+import org.gradle.util.GradleVersion
 
 /**
  * Validates required dependencies. If GPP can't run in the current context, an error will be
  * thrown.
  */
 fun Project.validateRuntime() {
-    rootProject.plugins.apply(RuntimeValidationPlugin::class)
-
-    project.plugins.withType<AppPlugin> {
+    GradleRuntimeValidator(GradleVersion.current(), MIN_GRADLE_VERSION).validate()
+    plugins.withType<AppPlugin> {
```

Comment:
I see no reason why this code had to live in a plugin nor why it had to only be applied to the root project. Either it will fail and then the build will stop or it will pass and then pass for every module this plugin is applied to.

Reply:
I think there were classpath issues back in the day, but probs doesn't matter anymore.

Reply:
Oh I see, but the Gradle version should be consistent across all modules?

Reply:
Yeah, I think this is fine. Did you see the build failure? That's what I'm waiting on.

Reply:
I did, it has been fixed now.